openmc/deplete/operator.py

```diff
@@ -720,103 +431,90 @@ def _unpack_tallies_and_normalize(self, source_rate):
-                number[i_nuc_results] = self.number[mat, nuc]
+        op_result = OperatorResult(keff, rates)
 
-            tally_rates = self._rate_helper.get_material_rates(
-                mat_index, nuc_ind, react_ind)
+        return copy.deepcopy(op_result)
 
-            # Compute fission yields for this material
-            fission_yields.append(self._yield_helper.weighted_yields(i))
+    def _update_materials(self):
+        """Updates material compositions in OpenMC on all processes."""
 
-            # Accumulate energy from fission
-            if fission_ind is not None:
-                self._normalization_helper.update(tally_rates[:, fission_ind])
+        for rank in range(comm.size):
+            number_i = comm.bcast(self.number, root=rank)
 
-            # Divide by total number and store
-            rates[i] = self._rate_helper.divide_by_adens(number)
+            for mat in number_i.materials:
+                nuclides = []
+                densities = []
+                for nuc in number_i.nuclides:
+                    if nuc in self.nuclides_with_data:
+                        val = 1.0e-24 * number_i.get_atom_density(mat, nuc)
 
-        # Scale reaction rates to obtain units of reactions/sec
-        rates *= self._normalization_helper.factor(source_rate)
+                        # If nuclide is zero, do not add to the problem.
+                        if val > 0.0:
+                            if self.round_number:
+                                val_magnitude = np.floor(np.log10(val))
+                                val_scaled = val / 10**val_magnitude
+                                val_round = round(val_scaled, 8)
 
-        # Store new fission yields on the chain
-        self.chain.fission_yields = fission_yields
+                                val = val_round * 10**val_magnitude
 
-        return OperatorResult(keff, rates)
+                            nuclides.append(nuc)
+                            densities.append(val)
+                        else:
+                            # Only output warnings if values are significantly
+                            # negative. CRAM does not guarantee positive
+                            # values.
+                            if val < -1.0e-21:
+                                print(f'WARNING: nuclide {nuc} in material'
+                                      f'{mat} is negative (density = {val}'
 
-    def _get_nuclides_with_data(self, cross_sections):
-        """Loads cross_sections.xml file to find nuclides with neutron data"""
-        nuclides = set()
-        data_lib = DataLibrary.from_xml(cross_sections)
-        for library in data_lib.libraries:
-            if library['type'] != 'neutron':
-                continue
-            for name in library['materials']:
-                if name not in nuclides:
-                    nuclides.add(name)
+                                      ' at/barn-cm)')
```

Comment:
To be consistent with usage elsewhere in the API
```suggestion
                                      ' atom/b-cm)')
```